### ISSUE TITLE
Duplicate AccountContact entity when contact changes from account

### ISSUE THREAD

**Issue by jasperPeeters98** [sulu/sulu#6447]
| Q | A
| --- | ---
| Sulu Version | 2.4.0
| PHP Version | 8.0
| DB Version | MySQL 5.7.36

# Actual Behavior

When the relation between a Sulu contact and a Sulu account has been changed, the previous relation does not get removed. This happens because the AccountContact does get removed but not flushed so when the contact is persisted, the relation has not been removed.

# Expected Behavior

When you change the relation, the previous relation should be removed.

# Steps to Reproduce

First you go to the contact section in the Sulu admin panel. Go to the edit form from a contact and change his organization.
<img width="1319" alt="Schermafbeelding 2022-01-07 om 10 04 35" src="https://user-images.githubusercontent.com/87987086/148519744-610fdf6e-1a05-4fb0-bdde-f4fd67084ff5.png">
When you go to the database, you can see that the relation has not been removed.
<img width="385" alt="Schermafbeelding 2022-01-07 om 10 03 31" src="https://user-images.githubusercontent.com/87987086/148519605-b08a54d2-1660-4b0c-a969-dbd192822c31.png">
When you go back to the contact section, there are now two contacts for the same contact.
<img width="1654" alt="Schermafbeelding 2022-01-07 om 10 06 01" src="https://user-images.githubusercontent.com/87987086/148519934-34df03f4-261e-4a5a-a5ff-80100d8aaa78.png">

# Possible Solutions

A possible solution is flushing right after the removal of the AccountContact. This means that the relation is deleted before persisting the contact. Here 'https://github.com/sulu/sulu/blob/2.x/src/Sulu/Bundle/ContactBundle/Contact/ContactManager.php#L613' it can look like this:
`$this->em->remove($mainAccountContact);`
`$this->em->flush();`

**Comment by alexander-schranz** [comment]
@jasperPeeters98 Thank you for report! This is really strange, flush is done at the end of the ContactManager::save method. I could think that we forget to remove it from the contact entity so it is persisted again. Can you check if something like the following would also work:

```php
    $contact->removeAccountContact($mainAccountContact);
    $this->em->remove($mainAccountContact);
```

If yes would you like to create a pull request for it?

This is related to the change in #6405 which was required for the Trash Handling. Do you know if it also happens for addresses or is there working all like expected, when remove an add new ones?

**Comment by jasperPeeters98** [comment]
@alexander-schranz Thank you for the response! I have checked the code that you suggested, this is indeed a better solution and i created a pull request for the fix. I have also tested the addresses and it seems that they work as expected.

**Comment by alexander-schranz** [comment]
fixed by #6448 thank you!
